README.md

```diff
@@ -652,6 +659,21 @@ was changed to simplify and streamline the process.
   The specific backup data location is now specified using a new required field: **`backup-data-path`**.
   This change allows you to reuse the same storage for different restore requests.
 
+### Backup Retention Policy
+
+This release introduces a new, configurable **RetentionPolicy** for managing backup storage more effectively.
+The feature allows users to specify retention rules for both full and incremental backups, replacing the previous rigid options (`KeepAll`, `RemoveAll`, `RemoveIncremental`).
+Retention policy is (optional) part of Backup Policy. It consists of 2 integer fields.
+
+* `full`: The total number of full backups to retain. If absent, all full backups are kept. Minimum is 1, it means that each new full backup will delete the previous one.
+* `incremental`: The number of full backups to retain their incremental backups. Cannot exceed `full`. Minimum is 0, it means that all old incremental backups will be deleted after each full. 
```

Comment:
I'm not clear exactly on how this parameter works, can you give more examples in a comment of what would happen with a value of 1, or what happens when it's the same as `full`? What happens if the user enters a value equal to the value of `full`?

Reply:
nil means no deletion.
full = nil, incr = nil => keep all
full = nil, incr = 0 => keep all full, delete all incremental after next full backup. So, incremental backups will actually happen, but they will live only up to next full backup.
full = 0 => forbidden
full = 1 => only store one full backup
full = N, incr = M => store N full backups, for last M of them store also incremental backups. N >= M.
full = N, incr = N => store N full backups, all with incrementals.

Does that make sense?

Reply:
Got it, I think I would rewrite it as such, please check if this is right:

`* `incremental`: The number of most recent full backups that also retain intermediate backups made between them. Cannot exceed the value of `full`. If omitted, all incremental backups are kept. A value of 0 means that all previous existing incremental backups will be deleted after each full backup is made.`

Please also add one or two lines saying what happens with the forbidden cases of "full = 0" and "incremental > full". Error? Fall back to default of nil?

Reply:
We never used word "intermediate" before, let's stick with "incremental".
In forbidden cases it will fail validation on startup. Just like any other field, we have a lot of validation rules.

Reply:
Yes, you're right of course on "incremental." I'll add something in the main docs about validation checks, no need to add it here. Copy in that wording and I'll re-review, thanks!